Add work remotely option in Create Post and Filter Page

aditilamkhede opened the pull request:
### What does this pull request aim to achieve? 💯
Add work remotely option in Create Post and in Filter on feed Page.

_Please be concise and link any related issue(s):_
adding a feature resolves issue #2180 
<!--### 🚨Before submitting this pull request🚨:-->

<!--_Please do **NOT** submit this PR if you have not done the following:_-->

- [x] I have checked that no one else is working on similar changes.
- [x] I have read the latest [**README**](README.md) and understand the development workflow.
- [x] The name of this branch is: **`feature/<branch_name>`**, or **`hotfix/<branch_name>`** if this is a hotfix, where `<branch_name>` briefly describes the issue(s) resolved and is prefixed with the issue number(s) (e.g. `1127-1130-update-git-branching-model`). The branch is created in a cloned version of this repo, **not a forked repo**.
- [x] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix).
- [x] There are no merge conflicts.
- [x] There are no console warnings and errors.
- [x] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".
- [x] The title describes the issue(s) being addressed using format: **`<issue_numbers> - <brief_issue_description>`** (e.g. `#1127/#1130 - Update Git Branching Model`)

aditilamkhede committed:
Add work reotely option in Create Post and Filter Page

Add work reotely option in Create Post and in Filter on feed Page.

aditilamkhede committed:
Merge branch 'staging' into feature/add-remote-CreatePostAndFilter-Pages

aditilamkhede committed:
added validation for remote selection

Added validation for remote dropdown value selection(i.e. making it required field) in CreatePost Form.
Added remote field in getPostsSchema.

aditilamkhede committed:
Change for older posts without remote

Change for older posts without remote option, i.e. database records/post which do not have remote option as it was added later. Make default option for those posts as 'No'.

aditilamkhede committed:
Changed removed hard code for default option

Changed removed hard code for default option from 'No' to options from createPostSettings.

aditilamkhede committed:
Changed Hard coded value

aditilamkhede committed:
Add and Hide Remote Work Label

hide the 'Remote Work' label from 'Create Post' modal.
For new post-> if user selects 'Yes' to qs 'can work remotely' then the post should have 'Remote Work' label

aditilamkhede committed:
Merge branch 'staging' into feature/add-remote-CreatePostAndFilter-Pages

aditilamkhede commented:
@rabbi08 here is the build URL: https://featureadd-remote-createpostandfilter-pa.fightpandemics.xyz/

aditilamkhede committed:
Merge branch 'staging' into feature/add-remote-CreatePostAndFilter-Pages

rabbi08 commented:
thanks @aditilamkhede 

below are few observations for this feature:

1. Please update label of 'Remote' as 'Remote Work' on Filters page.

2. The size, color and font of 'Remote option's "TEXT and ICON" should match with 'Share my location' option
Also great if we can align this option with 'Share my location' Please see below screenshot from desktop:
![image](https://user-images.githubusercontent.com/67168400/106176682-9edff280-6165-11eb-8348-f9c61e5fc308.png)

3. Search is performed when user click on 'Remote' option only. User should be able to select the checkbox and click on 'Apply Filters' to see the search results, just like for other filter options. This is to keep the search filters in sync, so that user can combine more than one filters.

4. After 'Remote' filter is applied and user is directed to help board, the 'applied filter option-remote' should be displayed to indicate the user that remote filter is in place, similar to other filter options. 

Please see below video for issue# 2 & 3:

https://user-images.githubusercontent.com/67168400/106177295-51b05080-6166-11eb-8885-81b86148714e.movPlease refer above video.

5. Filter: Existing post having tag as 'Remote Work' are NOT displayed in the search. Please see below video:

https://user-images.githubusercontent.com/67168400/106180275-17e14900-616a-11eb-9503-a2d20c2ee628.mov

6. Filtered Results: the post having no 'Remote work' tag is displayed. Please see below video:

https://user-images.githubusercontent.com/67168400/106181073-22e8a900-616b-11eb-9429-7e29e927bc2f.mov

aditilamkhede commented:
@rabbi08 I just have to work on 4th from above and then I will be done. 5 and 6are related to update data for old records. I will create script for old records and send it to @ddveloper.

aditilamkhede committed:
Apply/ClearFilter and add RemoteWork label to FilterList

Apply/ClearFilter functionality and added RemoteWork label to FilterList

aditilamkhede committed:
Removed Hardcoded text

aditilamkhede committed:
Merge branch 'staging' into feature/add-remote-CreatePostAndFilter-Pages

aditilamkhede committed:
Checkbox style

aditilamkhede committed:
Merge branch 'staging' into feature/add-remote-CreatePostAndFilter-Pages

aditilamkhede committed:
Update Remote feature with new Changes

Update Remote feature with new Changes. 3 Options on Create Person In-person, Remote, Both. Changes in filter on Helpborad page.

aditilamkhede committed:
Merge branch 'staging' into feature/add-remote-CreatePostAndFilter-Pages

aditilamkhede commented:
@rabbi08 Ready for testing. 
Here is the build url: https://featureadd-remote-createpostandfilter-pa.fightpandemics.xyz/feed

rabbi08 reviewed (changes requested):
Thanks @aditilamkhede 
Below are few initial observations, while testing this feature:
1. In Create Post window, please add a sentence below Question and align the dropdown with the visibility dropdown:
The Post can be fulfilled <dropdown>.

2. On HelpBoard -> Filter, Please update the filter heading 'Remote Work' as 'Post fulfillment' 

3. For Old Post having <Remote> tag 
    a. Are displayed as 'Both i.e. Remote and In- Person'  they should be 'Remote' not 'Both'
    b. and 'Remote' TAG is displayed on the post, we should NOT display the Remote tag on the post.
Please see below screenshot.
    
<img width="915" alt="image" src="https://user-images.githubusercontent.com/67168400/108021082-c8b16a00-6feb-11eb-939b-d3887453c479.png">

aditilamkhede commented:
@rabbi08 in addition to the 3rd observation from above,I believe we should remove "Remote Work" tag from "Types" filters as well on HelpBoard

rabbi08 commented:
> @rabbi08 in addition to the 3rd observation from above,I believe we should remove "Remote Work" tag from "Types" filters as well on HelpBoard

Yes plz
great catch @aditilamkhede 
Thanks

aditilamkhede committed:
Changes for old post and label changes

Changes for old post - post with 'Remote Work' tag should be displayed as 'Remote' if fullfillment is null else use fullfillment value.
post with no 'Remote Work' tag and fullfillment is null then display as "Both"
Label changes - Changed label text to Post Fullfillment on filter sidebar, added label on Create Post under Post fullfillment.

aditilamkhede commented:
@rabbi08 ready for review.

SinghRomika commented:
@aditilamkhede The question in the Create Post pop up modal window on the build  does not match the Feature description given on [Notion](https://www.notion.so/fightpandemics/Post-should-be-created-and-filtered-as-per-Remote-option-d1767e7eb31f49b59855c90c0573d206).
![Capture1](https://user-images.githubusercontent.com/78573508/108138294-8aab5900-7072-11eb-9b5b-40378097ea86.JPG)

Create Post pop up modal window on the build.
![Capture2](https://user-images.githubusercontent.com/78573508/108138569-158c5380-7073-11eb-875b-87bf489dffb7.JPG)

aditilamkhede committed:
Asiigned ids for gtm tag

Assigned ids to postfillment in filter section on help board and to fullfillment dropdown on create post page.

aditilamkhede commented:
> @aditilamkhede The question in the Create Post pop up modal window on the build does not match the Feature description given on [Notion](https://www.notion.so/fightpandemics/Post-should-be-created-and-filtered-as-per-Remote-option-d1767e7eb31f49b59855c90c0573d206).
> ![Capture1](https://user-images.githubusercontent.com/78573508/108138294-8aab5900-7072-11eb-9b5b-40378097ea86.JPG)
> 
> Create Post pop up modal window on the build.
> ![Capture2](https://user-images.githubusercontent.com/78573508/108138569-158c5380-7073-11eb-875b-87bf489dffb7.JPG)

@SinghRomika you are right, but notion needs to be updated. Notion is old. In current scenario there will be 3 options as you see currently.

SinghRomika commented:
@aditilamkhede  while reviewing the PR #2165 (new post fulfillment option in 'create post' and 'filter' page).
When i clicked the search my Location( Under Location). The location captured by the application was different than my original location.
(Location captured by the application)
![Filter Location ](https://user-images.githubusercontent.com/78573508/108252252-4b7e1600-710d-11eb-9bd8-cde017759d83.jpg)
(The difference between the Actual location vs what the application shows)
![Filter Location3_LI](https://user-images.githubusercontent.com/78573508/108252685-ccd5a880-710d-11eb-8af3-7b6502ae946a.jpg)

aditilamkhede commented:
> @aditilamkhede while reviewing the PR #2165 (new post fulfillment option in 'create post' and 'filter' page).
> When i clicked the search my Location( Under Location). The location captured by the application was different than my original location.
> (Location captured by the application)
> ![Filter Location ](https://user-images.githubusercontent.com/78573508/108252252-4b7e1600-710d-11eb-9bd8-cde017759d83.jpg)
> (The difference between the Actual location vs what the application shows)
> ![Filter Location3_LI](https://user-images.githubusercontent.com/78573508/108252685-ccd5a880-710d-11eb-8af3-7b6502ae946a.jpg)

Hi @SinghRomika not part of this PR. So, may be you want to file a bug. Also check the same on Staging before filing a bug.
@rabbi08 ?

SinghRomika commented:
@aditilamkhede I added this comment by mistake, please ignore it. I will check and review the Staging also.

SinghRomika commented:
@aditilamkhede Reviewed PR #2165 for now there seems to be no issue in the PR #2165 for desktop . Thanks!

rabbi08 commented:
Thanks @aditilamkhede 
The features looks great, the functionality is working as expected in Create post and 'filter' page.
I have an observation on mobile view:
- When user EDIT the post and have remote options value as both i.e. 'Remote & In-Person' the UI get distorted for the post i.e. the ellipses gets overlapped and the 'edited' text moved on the user profile icon. Please see below GIF and screenshot.

![IMG_2C0096AC6F77-1](https://user-images.githubusercontent.com/67168400/108541323-558c4a00-72b0-11eb-990d-5e6b1f95aa32.jpeg)
![ezgif com-video-to-gif-10](https://user-images.githubusercontent.com/67168400/108541517-8a000600-72b0-11eb-8d9a-a1266f390ce5.gif)

aditilamkhede commented:
> Thanks @aditilamkhede
> The features looks great, the functionality is working as expected in Create post and 'filter' page.
> I have an observation on mobile view:
> 
> * When user EDIT the post and have remote options value as both i.e. 'Remote & In-Person' the UI get distorted for the post i.e. the ellipses gets overlapped and the 'edited' text moved on the user profile icon. Please see below GIF and screenshot.
> 
> ![IMG_2C0096AC6F77-1](https://user-images.githubusercontent.com/67168400/108541323-558c4a00-72b0-11eb-990d-5e6b1f95aa32.jpeg)
> ![ezgif com-video-to-gif-10](https://user-images.githubusercontent.com/67168400/108541517-8a000600-72b0-11eb-8d9a-a1266f390ce5.gif)

@rabbi08 I will see what I can do. If there isn't any space available are we ok to trim some text or any other suggestions like decreasing font size?

rabbi08 commented:
> > Thanks @aditilamkhede
> > The features looks great, the functionality is working as expected in Create post and 'filter' page.
> > I have an observation on mobile view:
> > 
> > * When user EDIT the post and have remote options value as both i.e. 'Remote & In-Person' the UI get distorted for the post i.e. the ellipses gets overlapped and the 'edited' text moved on the user profile icon. Please see below GIF and screenshot.
> > 
> > ![IMG_2C0096AC6F77-1](https://user-images.githubusercontent.com/67168400/108541323-558c4a00-72b0-11eb-990d-5e6b1f95aa32.jpeg)
> > ![ezgif com-video-to-gif-10](https://user-images.githubusercontent.com/67168400/108541517-8a000600-72b0-11eb-8d9a-a1266f390ce5.gif)
> 
> @rabbi08 I will see what I can do. If there isn't any space available are we ok to trim some text or any other suggestions like decreasing font size?

Yes @aditilamkhede  we can trim for small size screens, if that work
Trim date can be like -
"Remote/In-pers" if that works

Thanks

ddveloper commented on client/src/components/CreatePost/Form/SecondSection.js:
```diff
@@ -4,7 +4,10 @@ import Head from "./Head";
-const types = Object.values(filterOptions)[2].options;
+let types = Object.values(filterOptions)[2].options;
+types = types.filter((val) => {
+  return val.value != "Remote Work";
```
may be use !== instead ?

aditilamkhede replied:
Done

ddveloper commented on client/src/components/Feed/FilterAccordion.js:
```diff
@@ -81,6 +137,8 @@ const FilterAccord = ({ gtmPrefix, locationOnly }) => {
     dispatchAction(SET_VALUE, "activePanel", activePanelKey);
   };
 
+  console.log("filters", filters, gtmPrefix);
```
remove console.log() ?

aditilamkhede replied:
Done

ddveloper commented on client/src/components/Feed/FilterAccordion.js:
```diff
@@ -114,22 +172,25 @@ const FilterAccord = ({ gtmPrefix, locationOnly }) => {
-                  {t(text)}
-                </ButtonTag>
-              );
+              console.log("idx", idx);
```
remove console.log() ?

aditilamkhede replied:
Done

ddveloper commented on client/src/components/Feed/Post.js:
```diff
@@ -604,6 +608,17 @@ const Post = ({
             <div className="pre-header post-page">
               <span>{t(`feed.${objective}`)}&nbsp;&nbsp;•</span>
+              <span>
+                &nbsp;&nbsp;
```
try <Space/> or other antd components? `&nbsp;` might not ideal

aditilamkhede replied:
tried using WhiteSpace - puts it into next line, antd Space doesn't help

ddveloper replied:
the [antd <Space>](https://ant.design/components/space/) might be useful here

ddveloper commented on client/src/components/Feed/Post.js:
```diff
@@ -741,6 +756,17 @@ const Post = ({
+                <span>
+                  &nbsp;&nbsp;
+                  {t(
+                    workMode && workMode !== "both"
```
This logic seems repeated several times, maybe extract it out

aditilamkhede replied:
Done

ddveloper commented on client/src/pages/Feed.js:
```diff
@@ -310,13 +316,38 @@ const Feed = (props) => {
     refetchPosts(null, null, true);
   };
 
+  /* //Aditi Do I need this */
```
maybe remove the commented code?

aditilamkhede replied:
Done

ddveloper commented on backend/lib/endpoints/posts.js:
```diff
@@ -113,6 +112,22 @@ async function routes(app) {
+      if (workMode){
+        const workModes = workMode.map(mode => mode.toLowerCase())
+        if(workModes.includes('both')){
+          console.log("In both")
```
remove the console.log() ?

aditilamkhede replied:
Done

ddveloper commented on backend/lib/endpoints/posts.js:
```diff
@@ -113,6 +112,22 @@ async function routes(app) {
+          filters.push({ $or: [ { $and: [ { workMode: null }, {types: { $nin : ['Remote Work'] } } ] }, {workMode: { $in: workModes } }] });
+        }
+        else if(workModes.includes('remote')) {
+          console.log("In remote")
```
remove the console.log() ?

aditilamkhede replied:
done

ddveloper commented on backend/lib/endpoints/posts.js:
```diff
@@ -113,6 +112,22 @@ async function routes(app) {
+        }
+        else if(workModes.includes('remote')) {
+          console.log("In remote")
+          filters.push({ $or: [ { $and: [ { workMode: null }, {types: { $in : ['Remote Work'] } } ] }, {workMode: { $in: workModes } }] });
```
This query seems complex, might be a better (simpler) way to achieve the goal ❔ 
it seems you are not using formatting tool Aditi, the format changes when I run prettier in VS code

aditilamkhede replied:
@ddveloper this one is basically adds logic to filter old records which don't have workMode fields but can have remote work tag in types feild.

aditilamkhede committed:
Changes for mobile view and refactor code.

aditilamkhede committed:
moved context menu (elipsis) to next line for mobile view

aditilamkhede committed:
Update Post.js

rabbi08 reviewed (approved):
Thanks @aditilamkhede 
Verified the fix on iPhone 11 Pro Max, this is working fine. Pleas see below screenshot. 
![image](https://user-images.githubusercontent.com/67168400/108926222-f1e58200-760b-11eb-8bab-8d793a7bf72c.jpeg)
LGTM

ddveloper reviewed (approved):
Thanks @aditilamkhede , LGTM! 👍
